src/model/deserialized.rs

```diff
@@ -52,8 +53,8 @@ pub struct BinXmlEntityReference<'a> {
 
 #[derive(Debug, PartialOrd, PartialEq, Clone)]
 pub struct BinXmlTemplate<'a> {
```

Comment:
This is kind of a bad name for what this struct actually is, since it doesn't make a lot of sense on its own anymore.
This would probably be better named reflecting the fact that it doesn't own the template data.

Maybe `BinXmlTemplateRef`